.github/workflows/listener.yml

```diff
@@ -18,90 +18,116 @@ jobs:
         id: replace-specs-snippets
-        with: 
+        with:
           branch-name: ${{ github.event.client_payload.branchName }}
           username: ${{ secrets.DX_GITHUB_USERNAME }}
           token: ${{ secrets.DX_GITHUB_TOKEN }}
           origin-repo: ${{ github.event.client_payload.originRepo }}
           author: ${{ github.event.client_payload.author }}
-      
+
       - name: Open Pull Request
         if: steps.replace-specs-snippets.outputs.needs_pr
         run: |
           hub pull-request --base Bandwidth:main $(if ${{ github.event.client_payload.draftPr }} ; then echo "-d" ; fi) -m '${{ github.event.client_payload.branchName }}' -m 'Update API specs from upstream api-specs repository. Opened By: @${{ github.event.client_payload.author }}'
-        env: 
+        env:
           GITHUB_TOKEN: ${{ secrets.DX_GITHUB_TOKEN }}
-      
+
       - name: Output PR Number
         run: |
           PR_NUMBER=$(hub pr list -h ${{ github.event.client_payload.branchName }} -f %I)
           echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_ENV
-        env: 
+        env:
           GITHUB_TOKEN: ${{ secrets.DX_GITHUB_TOKEN }}
 
       - name: Comment on Corresponding Repo
         uses: ./.github/actions/add_pr_comment
-        with: 
+        with:
           username: ${{ secrets.DX_GITHUB_USERNAME }}
           token: ${{ secrets.DX_GITHUB_TOKEN }}
           repo-name: ${{ github.event.client_payload.originRepo }}
           pr-number: ${{ github.event.client_payload.prNumber }}
           message: 'Preview site: http://bw-api-docs-${{ env.BRANCH_NAME }}.s3-website-us-east-1.amazonaws.com/\nPlease note that it may take a couple minutes for your preview site to become available.\n\nSee the corresponding PR opened on the docsite repository (no action required):\nhttps://github.com/Bandwidth/api-docs/pull/${{ env.PR_NUMBER }}'
 
-  merge: 
-    if: ${{ github.event.action == 'Merge' }} 
+  merge:
+    if: ${{ github.event.action == 'Merge' }}
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: '2'
+
       - name: Set PR number as env variable
         run: |
           echo "PR_NUMBER=$(hub pr list -h ${{ github.event.client_payload.branchName }} -f %I)" >> $GITHUB_ENV
-        env: 
+        env:
           GITHUB_TOKEN: ${{ secrets.DX_GITHUB_TOKEN }}
-      
-      - uses: ./.github/actions/add_pr_comment
-        with: 
-          username: ${{ secrets.DX_GITHUB_USERNAME }}
+
+      - name: Check Diff
+        run: |
+          echo "========== modified files =========="
+          git diff --name-only HEAD^ HEAD
+
+          echo "========== check paths of modified files =========="
+          git diff --name-only HEAD^ HEAD > files.txt
+          while IFS= read -r file
+          do
+            if [[ $file != site/specs/* ]]; then
+              echo "$file is not an API spec file. Can't automerge this PR."
```

Comment:
How is this going to react if we have a PR opened by api-specs that has a new spec file.  We would need to add additional changes, but this might result in the spec file being merged in without associated pages/changes being created.  How do we want to handle this?